src/bp/sdk/botpress.d.ts

```diff
@@ -433,104 +433,10 @@ declare module 'botpress/sdk' {
-      endpoint: string
-      authToken?: string
-    }
-
-    export interface Logger {
-      debug: (msg: string) => void
-      info: (msg: string) => void
-      warning: (msg: string, err?: Error) => void
-      error: (msg: string, err?: Error) => void
-    }
-
-    export interface TrainingSet {
-      intentDefs: NLU.IntentDefinition[]
-      entityDefs: NLU.EntityDefinition[]
-      languageCode: string
-      seed: number // seeds random number generator in nlu training
-    }
-
-    export interface ModelIdArgs extends TrainingSet {
-      specifications: Specifications
-    }
-
-    export interface TrainingOptions {
-      progressCallback: (x: number) => void
-      previousModel: ModelId | undefined
-    }
-
-    export interface Engine {
-      getHealth: () => Health
-      getLanguages: () => string[]
-      getSpecifications: () => Specifications
-      loadModel: (model: Model) => Promise<void>
-      unloadModel: (modelId: ModelId) => void
-      hasModel: (modelId: ModelId) => boolean
-      train: (trainSessionId: string, trainSet: TrainingSet, options?: Partial<TrainingOptions>) => Promise<Model>
-      cancelTraining: (trainSessionId: string) => Promise<void>
-      detectLanguage: (text: string, modelByLang: Dic<ModelId>) => Promise<string>
-      predict: (text: string, modelId: ModelId) => Promise<PredictOutput>
-      spellCheck: (sentence: string, modelId: ModelId) => Promise<string>
-    }
-
-    export const modelIdService: {
-      toString: (modelId: ModelId) => string // to use ModelId as a key
-      fromString: (stringId: string) => ModelId // to parse information from a key
-      toId: (m: Model) => ModelId // keeps only minimal information to make an id
-      isId: (m: string) => boolean
-      makeId: (factors: ModelIdArgs) => ModelId
-      briefId: (factors: Partial<ModelIdArgs>) => Partial<ModelId> // makes incomplete Id from incomplete information
-    }
-
-    export interface ModelId {
-      specificationHash: string // represents the nlu engine that was used to train the model
-      contentHash: string // represents the intent and entity definitions the model was trained with
-      seed: number // number to seed the random number generators used during nlu training
-      languageCode: string // language of the model
-    }
-
-    export interface Specifications {
-      nluVersion: string // semver string
-      languageServer: {
-        dimensions: number
-        domain: string
-        version: string // semver string
-      }
-    }
-
-    export type Model = ModelId & {
-      startedAt: Date
-      finishedAt: Date
-      data: {
-        input: string
-        output: string
-      }
-    }
-
-    export interface Health {
-      isEnabled: boolean
-      validProvidersCount: number
-      validLanguages: string[]
-    }
+    /**
+     * @experimental every types and functions from core are subject to change at any time.
+     */
+    export const core: typeof import('botpress/nlu')
```

Comment:
This line is pretty much the whole idea of this PR

Reply:
if we want to keep them types hidden from the user developer, Do we have to export them in the main Botpress typings file ?

Reply:
Well there is not only typings but implementations in nlu-core (makeEngine, modelIdService, etc...). I need those functions in nlu-module

Reply:
This is not a namespace, but just an exported variable so you can't use it like this:

```ts
let modelId: sdk.NLU.core.ModelId // won't build as core is a variable
```

```ts
let engine: sdk.NLU.core.makeEngine() // this works however
```

Reply:
What I'm trying to say here is why do you expose it in the sdk if you want to hide it from users. If it's because you need it in NLU module, you could simply build a "library" that you'll use both in STAN and in NLU module.

Check botpress/common you'll see what I mean